src/probe_matcher.cpp

```diff
@@ -115,7 +115,10 @@ std::set<std::string> ProbeMatcher::get_matches_for_probetype(
     case ProbeType::kprobe:
     case ProbeType::kretprobe:
     {
-      symbol_stream = get_symbols_from_traceable_funcs();
+      if (!target.empty())
+        symbol_stream = get_symbols_from_traceable_funcs(true);
+      else
+        symbol_stream = get_symbols_from_traceable_funcs(false);
```

Comment:
```suggestion
      symbol_stream = get_symbols_from_traceable_funcs(!target.empty());
```

Reply:
I think this makes the intention of the code slightly more difficult to read, even if it's a few more lines, so I'd prefer the original.